feature/qr-test-doc (#2304)
mercadopago/devsite-docs: guides/qr-code/qr-attended-test-purchase.pt.md

> # Fazer uma compra teste 

Reviewer: @pcanosa - "Fazer" acaba ficando informalzão demais. Seguimos com Realizar, aí é só ajustar no menu também. O que acha?

> A seguir, vamos te mostrar como simular um fluxo de pagamento completo para o Código QR. 

Reviewer: ```suggestion
Siga as etapas descritas abaixo para simular um fluxo de pagamento completo para Código QR.
```

> WARNING
>
> Importante
>

> > Durante todo o fluxo de teste, você deverá usar as **credenciais de produção dos usuários de teste** que você criou anteriormente. Em cada passo, será indicado se é o usuário vendedor ou comprador.

Reviewer: ```suggestion
> Em cada etapa do fluxo de teste, utilize as **credenciais de produção dos usuários de teste** previamente criadas. Em cada etapa, será indicado se é o usuário vendedor ou comprador.
```

> > Durante todo o fluxo de teste, você deverá usar as **credenciais de produção dos usuários de teste** que você criou anteriormente. Em cada passo, será indicado se é o usuário vendedor ou comprador.

Author: ```suggestion
> Durante todo o fluxo de teste, você deverá usar as **credenciais de produção dos usuários de teste** previamente criadas. Em cada etapa, será indicado se é o usuário vendedor ou comprador.
```

## Criar pedido 

> 1. Faça login **no site do Mercado Pago** com o nome de usuário e a senha da conta de vendedor teste que você criou. 

Reviewer: @pcanosa - Pili, em "site do Mercado Pago", poderia ser interessante já inserir o link para a página, não?

> 1. Faça login **no site do Mercado Pago** com o nome de usuário e a senha da conta de vendedor teste que você criou. 

Reviewer: ```suggestion
1. Faça login **no site do Mercado Pago** com o nome de usuário e a senha da conta de vendedor teste previamente criados. 
```

2. Crie uma [nova aplicação](/developers/pt/docs/qr-code/additional-content/your-integrations/dashboard) de teste para o código QR e obtenha as credenciais de produção (Access Token) do usuário de vendedor de teste. 
3. Use as credenciais de produção do usuário de vendedor de teste para criar uma [loja](/developers/pt/reference/stores/_users_user_id_stores/post) e uma [caixa](/developers/pt/reference/pos/_pos/post), seguindo as instruções para configurá-las. Tenha em mente que, ao criar a caixa, o campo `fixed_amount` deve ser definido como `true`. Os links retornados na resposta para o objeto `qr` são as imagens do Código QR associadas à sua caixa. 

> 4. Ainda usando as credenciais do usuário de vendedor de teste, [crie um pedido](/developers/pt/reference/instore_orders_v2/_instore_qr_seller_collectors_user_id_stores_external_store_id_pos_external_pos_id_orders/put) e atribua-o à caixa que você criou no passo anterior. Ao fazer isso, você também estará atribuindo esse pedido ao Código QR associado à caixa. Certifique-se de configurar o campo `notification_url` com a URL onde você receberá as notificações sobre as atualizações de pagamentos com o tópico `merchant_order`. 

Reviewer: ```suggestion
4. Ainda com as **credenciais do usuário vendedor de teste**, [crie um pedido](/developers/pt/reference/instore_orders_v2/_instore_qr_seller_collectors_user_id_stores_external_store_id_pos_external_pos_id_orders/put) e atribua-o à caixa criada no passo anterior. Este processo também atribui o pedido ao Código QR da caixa. Configure o campo `notification_url` para a URL onde deseja receber notificações sobre atualizações de pagamentos, utilizando o tópico `merchant_order`.
```

## Realizar pagamento 

> 1. Baixe e instale **o aplicativo do Mercado Pago em seu dispositivo móvel** e faça login com a conta do usuário de teste definido como comprador. 

Reviewer: ```suggestion
1. Baixe e instale **o aplicativo do Mercado Pago em seu dispositivo móvel** e faça login com a conta do usuário de teste definido como comprador. 
```

> 2. Escaneie do seu dispositivo móvel o Código QR que você gerou. O aplicativo mostrará o valor do pedido gerado e as opções de pagamento disponíveis. 

Reviewer: ```suggestion
2. Utilize seu dispositivo móvel para escanear o Código QR gerado anteriormente. O aplicativo exibirá o valor do pedido e as opções de pagamento disponíveis.
```

> 3. Realize o pagamento usando [cartões de teste](/developers/pt/docs/qr-code/additional-content/your-integrations/test/cards), que também permitirão que você teste diferentes fluxos de pagamento. 

Reviewer: ```suggestion
3. Realize o pagamento utilizando [cartões de teste](/developers/pt/docs/qr-code/additional-content/your-integrations/test/cards), que também permitirão testar diferentes fluxos de pagamento. 
```

### Casos de validação 

> Se você desejar, pode testar diferentes casos para validar se o seu sistema está integrado corretamente com o Mercado Pago. Estes são cenários que simulam situações que podem ocorrer ao realizar um pagamento. 

Reviewer: ```suggestion
Caso queira, é possível testar diferentes casos para validar se o seu sistema está integrado corretamente com o Mercado Pago. Estes cenários simulam situações que podem ocorrer ao realizar um pagamento. 
```

> A seguir, você pode ver estes casos em detalhes, o resultado esperado no sistema para cada situação, e uma série de observações que indicarão o que fazer em cada caso.

Reviewer: ```suggestion
A seguir, estão disponíveis detalhes sobre cada caso, incluindo o resultado esperado no sistema para diferentes situações e orientações sobre como proceder em cada cenário.
```

| Caso | Resultado esperado | Observações |
|---|---|---|
| **Escaneamento correto do QR**. O usuário escaneia um Código QR válido, ou seja, um QR que foi previamente associado a um pedido.  | O aplicativo exibe a tela de pagamento com o valor atribuído. | Verifique se o valor exibido na tela de pagamento corresponde ao valor atribuído durante a criação da ordem. Caso não corresponda, verifique se a ordem foi corretamente atribuída à caixa correta. |

> | **Pagamento aprovado**. O usuário realiza um pagamento e ele é aprovado. | O sistema de Ponto de Venda recebe as informações de um pagamento aprovado. | Verifique se você recebeu as [notificações](/developers/pt/docs/qr-code/additional-content/your-integrations/notifications) com o tópico `merchant_order` e verifique se o status delas está como `closed`. |

Reviewer: ```suggestion
| **Pagamento aprovado**. O usuário realiza um pagamento e ele é aprovado. | O sistema do PDV (Ponto de Venda) recebe as informações de um pagamento aprovado. | Verifique se você recebeu as [notificações](/developers/pt/docs/qr-code/additional-content/your-integrations/notifications) com o tópico `merchant_order` e verifique se o status delas está como `closed`. |
```

| **Cancelar pedido**. O usuário muda de ideia e decide pagar em dinheiro, por exemplo. | [Elimine o pedido](/developers/pt/reference/instore_orders_v2/_instore_qr_seller_collectors_user_id_pos_external_pos_id_orders/delete) da caixa. | Escaneie o QR e verifique se a tela de espera é exibida. Isso confirmará que o pedido foi removido corretamente. |
| **Escaneamento prévio à criação do pedido**. O cliente escaneia um código QR válido antes de atribuir-lhe o pedido. | O aplicativo mostra a tela de espera. | Verifique se você configurou o campo `fixed_amount=true` ao criar a caixa. |
| **Pagamento recusado**. O usuário realiza um pagamento e ele é recusado. | O sistema de Ponto de Venda recebe as informações de um pagamento recusado e continua esperando que o pedido seja pago. | Verifique se você recebeu as [notificações](/developers/pt/docs/qr-code/additional-content/your-integrations/notifications) com o tópico `merchant_order` e confirme se o status dessas notificações está como `opened`. Aguarde a segunda tentativa de pagamento. |

> | **Segunda tentativa de pagamento**. O pagamento realizado pelo usuário é rejeitado e, sem a necessidade de escanear novamente o QR, uma segunda tentativa é feita e é aprovada. | O sistema do Ponto de Venda recebe as informações de um pagamento rejeitado e, em seguida, de um pagamento aprovado. | Não elimine o pedido após um pagamento recusado e verifique que o seu Ponto de Venda não encerra a transação na caixa.  |

Reviewer: ```suggestion
| **Segunda tentativa de pagamento**. O pagamento realizado pelo usuário é rejeitado e, sem a necessidade de escanear novamente o QR, uma segunda tentativa é feita e é aprovada. | O sistema do Ponto de Venda recebe as informações de um pagamento rejeitado e, em seguida, de um pagamento aprovado. | Não elimine o pedido após um pagamento recusado e garanta que seu Ponto de Venda não encerre a transação na caixa.  |
```

> WARNING
>
> Importante
>
> Apesar de o ambiente de testes não permitir validar casos de devolução de pagamentos ou reembolsos, recomendamos que você implemente o fluxo de devoluções na sua integração de produção utilizando nossa [API de reembolsos](/developers/pt/reference/chargebacks/_payments_id_refunds/post).

## Verificar notificações 

> Após ter realizado o pagamento com o usuário de teste, verifique se você recebeu as notificações com o tópico `merchant_order` sobre os estados do pedido em seu sistema. 

Reviewer: ```suggestion
Após a realização do pagamento com o usuário de teste, verifique o recebimento das notificações com o tópico `merchant_order` sobre os estados do pedido em seu sistema. 
```

> Para verificar se elas foram processadas corretamente, faça uma requisição **GET** para o endpoint [/merchant_orders](/developers/pt/reference/merchant_orders/_merchant_orders_id/get) com o ID da merchant order que você recebeu na notificação. Isso permitirá que você valide o estado de cada uma: 

Reviewer: ```suggestion
Para verificar se elas foram processadas corretamente, faça uma requisição **GET** para o endpoint [/merchant_orders](/developers/pt/reference/merchant_orders/_merchant_orders_id/get) com o ID da merchant order recibida na notificação. Isso permitirá validar o estado de cada uma: 
```

4. Ainda com as **credenciais do usuário vendedor de teste**, [crie um pedido](/developers/pt/reference/instore_orders_v2/_instore_qr_seller_collectors_user_id_stores_external_store_id_pos_external_pos_id_orders/put) e atribua-o à caixa criada no passo anterior. Este processo também atribui o pedido ao Código QR da caixa. Configure o campo `notification_url` para a URL onde deseja receber notificações sobre atualizações de pagamentos, utilizando o tópico `merchant_order`.

## Realizar pagamento 

1. Baixe o aplicativo do Mercado Pago no seu dispositivo móvel, disponível para [Android](https://play.google.com/store/apps/details?id=com.mercadopago.wallet&pcampaignid=web_share) e [iOS](https://apps.apple.com/br/app/mercado-pago-banco-digital/id925436649), instale-o e, em seguida, faça login com as **credenciais da conta de usuário de teste definida como comprador**.

> 2. Utilize seu dispositivo móvel para escanear o Código QR gerado anteriormente. O aplicativo exibirá o valor do pedido e as opções de pagamento disponíveis.

Author: ```suggestion
2. Utilize seu dispositivo móvel para escanear o Código QR gerado anteriormente. O aplicativo exibirá o valor do pedido e as opções de pagamento disponíveis.

> WARNING
>
> Importante
>
>  Os pagamentos Pix não podem ser feitos com credenciais de teste. Recomendamos selecionar outra forma de pagamento para realizar o teste de integração do Código QR.
------------
```

4. Ainda com as **credenciais do usuário vendedor de teste**, [crie um pedido](/developers/pt/reference/instore_orders_v2/_instore_qr_seller_collectors_user_id_stores_external_store_id_pos_external_pos_id_orders/put) e atribua-o à caixa criada no passo anterior. Este processo também atribui o pedido ao Código QR da caixa. Configure o campo `notification_url` para a URL onde deseja receber notificações sobre atualizações de pagamentos, utilizando o tópico `merchant_order`.

## Realizar pagamento 

1. Baixe o aplicativo do Mercado Pago no seu dispositivo móvel, disponível para [Android](https://play.google.com/store/apps/details?id=com.mercadopago.wallet&pcampaignid=web_share) e [iOS](https://apps.apple.com/br/app/mercado-pago-banco-digital/id925436649), instale-o e, em seguida, faça login com as **credenciais da conta de usuário de teste definida como comprador**.
2. Utilize seu dispositivo móvel para escanear o Código QR gerado anteriormente. O aplicativo exibirá o valor do pedido e as opções de pagamento disponíveis.

> > WARNING

Author: ```suggestion
----[mlb]----
> WARNING
```

>
> Importante
>

> >  Os pagamentos Pix não podem ser feitos com credenciais de teste. Recomendamos selecionar outra forma de pagamento para realizar o teste de integração do Código QR.

Author: ```suggestion
>  Os pagamentos Pix não podem ser realizados com credenciais de teste. Recomendamos selecionar outra forma de pagamento para realizar o teste de integração do Código QR.
```
